heareval/task_embeddings.py

```diff
@@ -22,102 +22,117 @@
 
     def __getitem__(self, idx):
-        return self.rows[idx]
+        r = self.rows.iloc[idx]
+        return r["slug"]
 
 
 # High-level wrapper on the basic API that we might
 # consider sharing among all participants.
 def get_audio_embedding_numpy(
     audio_numpy: np.ndarray,
     model: Any,
-    frame_rate: float,
-) -> Tuple[Dict[int, np.ndarray], np.ndarray]:
-    embedding, timestamps = EMBED.get_audio_embedding(  # type: ignore
-        torch.tensor(audio_numpy, device=device),
-        model=model,
-        frame_rate=frame_rate,
-    )
-    embedding = embedding.detach().cpu().numpy()
-    timestamps = timestamps.detach().cpu().numpy()
-    return embedding, timestamps
+    task_type: str,
+) -> Tuple[Dict[int, np.ndarray], Optional[np.ndarray]]:
+    if task_type == "scene_labeling":
+        embeddings = EMBED.get_scene_embeddings(  # type: ignore
+            torch.tensor(audio_numpy, device=device),
+            model=model,
+        )
+        embeddings = embeddings.detach().cpu().numpy()
+        return embeddings, None
+    elif task_type == "event_labeling":
+        embeddings, timestamps = EMBED.get_timestamp_embeddings(  # type: ignore
+            torch.tensor(audio_numpy, device=device),
+            model=model,
+        )
+        embeddings = embeddings.detach().cpu().numpy()
+        timestamps = timestamps.detach().cpu().numpy()
+        return embeddings, timestamps
+    else:
+        raise ValueError(f"Unknown task_type = {task_type}")
 
 
 def task_embeddings():
-    model = EMBED.load_model(EMBEDDING_MODEL_PATH, device=device)  # type: ignore
+    model = EMBED.load_model(EMBEDDING_MODEL_PATH)
 
     # TODO: Would be good to include the version here
     # https://github.com/neuralaudio/hear2021-eval-kit/issues/37
     embeddir = os.path.join("embeddings", EMBED.__name__)  # type: ignore
     embed_sr = model.sample_rate
 
+    # TODO: tqdm with task name
     for task in glob.glob("tasks/*"):
-        # TODO: We should be reading the metadata that describes
-        # the frame_rate.
-        # https://github.com/neuralaudio/hear2021-eval-kit/issues/53
-        frame_rate = 10
-
-        # TODO: Include "val" ?
-        for split in ["train", "test"]:
+        print(f"Task {task}")
+        task_config = json.loads(open(os.path.join(task, "task_metadata.json")).read())
+        for split in ["train", "valid", "test"]:
             print(f"Getting embeddings for {split} split:")
 
+            metadata_csv = os.path.join(task, f"{split}.csv")
+            if not os.path.exists(metadata_csv):
+                continue
+
             # TODO: We might consider skipping files that already
             # have embeddings on disk, for speed
+            # TODO: Choose batch size based upon audio file size?
+            # Or assume that the API is smart enough to do this?
+            # How do we test for memory blow up etc?
+            # e.g. that it won't explode on 10 minute audio
```

Comment:
Ya good question -- GPU memory management should be handled by the API right? Don't we demonstrate using a batch size to send to the model in a framewise way? Can we just load the audio into CPU memory and assume that the models are going to load them into GPU memory in a smart way?

Reply:
https://github.com/neuralaudio/hear-eval-kit/issues/101